pkg/virtctl/imageupload/imageupload.go

```diff
@@ -926,3 +951,56 @@ func handleEventErrors(client kubecli.KubevirtClient, pvcName, dvName, namespace
 
 	return nil
 }
+
+func createNewDataSource(client kubecli.KubevirtClient, name, namespace string, labels map[string]string) error {
+	ds := &v1beta1.DataSource{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels:    labels,
+		},
+		Spec: v1beta1.DataSourceSpec{
+			Source: v1beta1.DataSourceSource{
+				PVC: &v1beta1.DataVolumeSourcePVC{
+					Name:      name,
+					Namespace: namespace,
+				},
+			},
+		},
+	}
+
+	if _, err := client.CdiClient().CdiV1beta1().DataSources(namespace).Create(context.Background(), ds, metav1.CreateOptions{}); err != nil {
+		return err
+	}
+
+	fmt.Printf("Created new DataSource %s/%s\n", namespace, name)
+	return nil
+}
+
+func updateExistingDataSource(client kubecli.KubevirtClient, name, namespace string, labels map[string]string) error {
```

Comment:
```suggestion
func updateExistingDataSource(client kubecli.KubevirtClient, ds *cdiv1.DataSource) error {
```

Make it take the existing DS as an argument and work with it inside this func.

Reply:
It gets the namespace of the PVC, not of the DS. I'll udpate it.

Reply:
I'm not sure about it, because `name` and `namespace` refers to PVC object, here:
```
patch.PatchOperation{
	Op:   patch.PatchReplaceOp,
	Path: "/spec/source/pvc",
	Value: map[string]string{
		"name":      name,
		"namespace": namespace,
	},
},
```
And `labels` refers to the labels of the DataSource.

Reply:
Potentially only `labels` can be replaced by `ds *cdiv1.DataSource`.

Reply:
I've changed this to:
```
func updateExistingDataSource(client kubecli.KubevirtClient, pvcName, pvcNamespace string, ds *cdiv1.DataSource) error {
```